src/cs50/sql.py

```diff
@@ -22,12 +24,52 @@ def __init__(self, url, **kwargs):
         http://docs.sqlalchemy.org/en/latest/dialects/index.html
         """
 
-        # log statements to standard error
+        # Require that file already exist for SQLite
+        matches = re.search(r"^sqlite:///(.+)$", url)
+        if matches:
+            if not os.path.exists(matches.group(1)):
+                raise RuntimeError("does not exist: {}".format(matches.group(1)))
+            if not os.path.isfile(matches.group(1)):
+                raise RuntimeError("not a file: {}".format(matches.group(1)))
+
+        # Create engine, raising exception if back end's module not installed
+        self.engine = sqlalchemy.create_engine(url, **kwargs)
+
+        # Log statements to standard error
         logging.basicConfig(level=logging.DEBUG)
         self.logger = logging.getLogger("cs50")
 
-        # create engine, raising exception if back end's module not installed
-        self.engine = sqlalchemy.create_engine(url, **kwargs)
+        # Test database
+        try:
+            self.logger.disabled = True
+            self.execute("SELECT 1")
+        except sqlalchemy.exc.OperationalError as e:
+            e = RuntimeError(self._parse(e))
+            e.__cause__ = None
+            raise e
+        else:
+            self.logger.disabled = False
+
+    def _parse(self, e):
+        """Parses an exception, returns its message."""
+
+        # MySQL
+        matches = re.search(r"^\(_mysql_exceptions\.OperationalError\) \(\d+, \"(.+)\"\)$", str(e))
+        if matches:
+            return matches.group(1)
+
+        # PostgreSQL
+        matches = re.search(r"^\((psycopg2\.OperationalError)\) (.+)$", str(e))
+        if matches:
+            return matches.group(1)
```

Comment:
Would that return `psycopg2.OperationalError` instead of the actual error message?